mapbox/dr-ui
# [info arch] Update Breadcrumbs on mobile

This PR makes some updates to the Breadcrumbs component and how PageLayout displays the Breadcrumb component:

- On mobile, Breadcrumbs will display the second to last link with a back arrow. (Ignore the dotted border, that's from the test cases app).

Device size | Image
---|---
Large | ![image](https://user-images.githubusercontent.com/2180540/98958427-425b2880-24d0-11eb-9b04-52015fcff427.png)
Small | ![image](https://user-images.githubusercontent.com/2180540/98958388-38392a00-24d0-11eb-8970-41b8ecd128a6.png)

- While testing, I noticed that Breadcrumbs should only be shown on mobile if there is no sidebar. Otherwise the breadcrumbs site underneath the navigation and is redundant. This PR will make it so that the Breadcrumb component in PageLayout will only display on mobile if there is no sidebar. This is helpful in sites like Playground which we project will not have a sidebar, so the Breadcrumbs will be vital in allowing the user to return the the Playground homepage from a given playground.

- While testing, I also noticed the the Breadcrumbs on multi-structured sites were a little misleading. This PR will update PageLayout’s Breadcrumb data for multi-structured sites to **not** show the site name. For multi-structured sites, the site name is not an independent page and redirects to one of the subsides. For example /android redirects to /android/maps. In the future, if we decide to add platform pages, then we can update the crumbs. In the images below, if you are on the Navigation SDK Overview page and click "Android" it will send you to Maps SDK for Android -- which would likely be confusing to the user.

Stage | Image
---|---
Current | ![image](https://user-images.githubusercontent.com/2180540/98958580-6b7bb900-24d0-11eb-991e-2058bbb21dc4.png)
Proposed | ![image](https://user-images.githubusercontent.com/2180540/98958603-720a3080-24d0-11eb-9a01-4c739bc3f723.png)


## How to test

- Observe the /Breadcrumb test cases app
- Observe the /PageLayout test cases, including "No sidebar, show Breadcrumbs on mobile." 

## QA checklist




Open the test cases app locally on:
